assets/extensions/main.js

```diff
@@ -6,21 +13,42 @@ import Tabs from './tabs';
+
+const Main = () => {
+	const [ extensions, setExtensions ] = useState( false );
+
+	useEffect( () => {
+		// TODO: Update to real endpoint.
+		apiFetch( { path: '/sensei-internal/v1/setup-wizard/features' } ).then(
+			() => {
+				setExtensions( extensionsMock );
+			}
+		);
+	}, [] );
+
+	if ( false === extensions ) {
+		return (
+			<div className="sensei-extensions__loader">
+				<Spinner />
```

Comment:
The spinner looks a bit too small for the page. Did you investigate if it's possible to increase it's size? With a quick look I think that it is a bit hard to accomplish. We could potentially use WooCommerce's spinner which is easier to modify.

Reply:
> The spinner looks a bit too small for the page. Did you investigate if it's possible to increase it's size? With a quick look I think that it is a bit hard to accomplish.

I agree that it's very small considering the whole page. But there's no prop to change the size. It gets from a const with configurations. And injecting it through CSS would be hard because of the way this Spinner works.

> We could potentially use WooCommerce's spinner which is easier to modify.

About WooCommerce, I don't think it would be a good idea to increase our dependencies only for the spinner because it'll not appear when we add the preload. Maybe if we have other cases in the future, a good idea would be to create our own loader. Maybe something using the sensei Logo? 😀

Reply:
> Maybe something using the sensei Logo?

We could make a bonsai treei that is growing 🙂 

>  I don't think it would be a good idea to increase our dependencies only for the spinner because it'll not appear when we add the preload

I agree that we shouldn't bother if we intend to preload the page. On a more general note however, is increasing our dependencies in the editor an issue? I mean we should definitely look into not adding dependencies on the frontend but in the editor the traffic is very small, would it make much difference?

Reply:
The API result for the extensions can be preloaded, so this would appear immediately — no spinner needed :)